dart/lib/src/hub.dart

```diff
@@ -0,0 +1,280 @@
+  SentryId get lastEventId => _lastEventId;
+
+  /// Captures the event.
+  Future<SentryId> captureEvent(Event event) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureEvent' call is a no-op.",
+      );
+    } else if (event == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureEvent called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureEvent(event, scope: item.scope);
+        } catch (err) {
+          /* TODO add Event.id */
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing event with id: ${event}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureEvent',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  /// Captures the exception
+  Future<SentryId> captureException(
+    dynamic throwable, {
+    dynamic stackTrace,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureException' call is a no-op.",
+      );
+    } else if (throwable == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureException called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          // TODO pass the scope
+          sentryId = await item.client.captureException(
+            throwable,
+            stackTrace: stackTrace,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing exception : ${throwable}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureException',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  /// Captures the message.
+  Future<SentryId> captureMessage(
+    Message message, {
+    SeverityLevel level = SeverityLevel.info,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureMessage' call is a no-op.",
+      );
+    } else if (message == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureMessage called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          // TODO pass the scope
+          sentryId = await item.client.captureMessage(message, level: level);
+        } catch (err) {
+          _options.logger(
```

Comment:
Here we clearly failed to capture, so again I wouldn't suggest assigning the eventId. Let's just do that if we at least were able to pass the event to the client?

Reply:
the `_lastEventId = sentryId;` assignment ?

Reply:
ok, so if we were able to call captureMessage or captureEvent... we assign the sentryId/lastEventId otherwise it really doesnt make sense, this should be fixedi n Java too